### PR TITLE
fix(nado): order nonce entropy, trigger wrappers, param handling

### PR DESCRIPTION
**Pull request by rayBastard**
Adds random low bits to order nonces, auto-selects the trigger endpoint in fetchCanceledOrders/fetchCanceledAndClosedOrders, forwards limit in trigger order wrappers, normalizes triggerDirection, caps archive limits at 500 and rejects trigger params in editOrder.

### CODE REVIEW

File: ts/src/nado.ts [hunk 1]
```diff
@@ -977,7 +982,7 @@ export default class nado extends Exchange {
             'product_ids': productIds,
         };
         if (limit !== undefined) {
-            request['limit'] = limit;
+            request['limit'] = Math.min (limit, 500);
```
Comment (carlotestor) [inline ts/src/nado.ts:986]: The `fetchOHLCV` cap got a dedicated regression fixture ("fetchOHLCV caps the limit at the maximum of 500"), but this one didn't -- every `fetchOrders`/wrapper fixture passes `limit: 10`, so nothing actually exercises this clamp. A fourth entry with `limit: 1000` expecting `"limit": 500` would pin both caps symmetrically and keep them from silently regressing.

File: ts/src/nado.ts [hunk 2]
```diff
@@ -550,6 +551,10 @@ export default class nado extends Exchange {
         if (type !== 'limit') {
             throw new InvalidOrder (this.id + ' editOrder() supports limit orders only');
         }
+        const triggerPrice = this.safeStringN (params, [ 'triggerPrice', 'stopPrice', 'stopLossPrice', 'takeProfitPrice' ]);
+        if (triggerPrice !== undefined) {
+            throw new NotSupported (this.id + ' editOrder() does not support trigger orders, cancel the trigger order and create a new one instead');
+        }
```
Comment (carlotestor) [inline ts/src/nado.ts:558]: This guard sits in `editOrderRequest`, which `editOrderWs` also calls (`ts/src/pro/nado.ts:855`), so a WS caller passing `triggerPrice` gets an error message naming `editOrder()`. Worth widening the text to cover both entry points. Separately, the `nado#editOrder` JSDoc doesn't mention the new rejection -- since four param keys are now hard-refused, a line documenting that trigger params are unsupported would save a round trip.

File: ts/src/nado.ts [hunk 3]
```diff
@@ -2913,7 +2919,12 @@ export default class nado extends Exchange {
 
     createOrderNonce (recvWindow: any) {
         const expires = this.sum (this.milliseconds (), recvWindow);
-        return Precise.stringMul (this.numberToString (expires), '1048576');
+        const highBits = Precise.stringMul (this.numberToString (expires), '1048576');
+        // the exchange defines the nonce to be the recv time moved left by 20 bits
+        // plus 20 random low bits, otherwise two orders created during the same
+        // millisecond would collide on the same nonce and get rejected
+        const entropy = this.randNumber (6);
+        return Precise.stringAdd (highBits, this.numberToString (entropy));
```
Comment (carlotestor) [inline ts/src/nado.ts:2928]: Minor wording: `randNumber (6)` yields 0-999999, which covers ~95.4% of the 20-bit field rather than a full "20 random low bits" as the comment says. The behaviour is fine -- it can't overflow into the recvTime bits, and it's strictly more entropy than Nado's own `gen_order_nonce` (`randint(0, 999)`) -- so this is just the comment being slightly more precise than the code.

**Review by carlotestor — COMMENTED** [review]
## Summary

Checked the three claimed fixes against Nado's own docs and SDK — all three are real bugs on `master`, not cosmetic.

**Trigger wire key.** `master` built `'oracle_price_' + triggerDirection` straight from user input, so `triggerDirection: 'up'` shipped `oracle_price_up`. Nado's SDK only defines `oracle_price_above` / `oracle_price_below` (plus the `last_price_*` / `mid_price_*` pairs), so every `up`/`above` trigger order went out with an unrecognised key. Routing through `handleTriggerDirectionAndParams` fixes that and picks up the unified `ascending`/`descending` vocabulary at the same time. I confirmed the helper still rejects a missing or bogus direction, so the `ArgumentsRequired` this PR deletes is not lost — it moved to the base:

```
{}                           -> THROWS ArgumentsRequired: trigger orders require params["triggerDirection"] ...
{"triggerDirection":"up"}    -> ascending   -> oracle_price_above
{"triggerDirection":"below"} -> descending  -> oracle_price_below
{"triggerDirection":"sideways"} -> THROWS ArgumentsRequired
```

**Nonce entropy.** Nado's `gen_order_nonce` is `(recv_time_ms << 20) + random_int`. `master` did the shift and added nothing, so two orders created in the same millisecond collided on an identical nonce. This matches the reference implementation.

**Archive caps.** Vendor docs confirm `limit` maxes at 500 for both candlesticks and `list_trigger_orders`, so both `Math.min (limit, 500)` calls are correct and consistent with the six that already exist in this file. Dropping `limit` in the trigger wrappers was silently handing callers the server default of 100 instead of what they asked for.

Worth noting the `fetchCanceledOrders` / `fetchCanceledAndClosedOrders` change is a bigger deal than the title suggests: both previously hit `fetchOrders`' `nado fetchOrders only support trigger` throw unless the caller hand-passed `trigger: true`, so they were unusable through the documented signature. The three corrected `"method"` fields in the fixture (`fetchCanceledAndClosedOrders` -> `createOrder`) were also mislabelling which method those createOrder cases actually exercised.

### Verification on `8698c70`

| check | result |
|---|---|
| `npm run tsBuild` | pass |
| eslint `ts/src/nado.ts` + `ts/src/pro/nado.ts` | clean, exit 0 |
| `--requestTests nado` (JS) | 43/43, identical across 3 consecutive runs |
| `--requestTests nado` (Python, async) | 43/43 |
| `master` baseline, same tree | 40/40 -> +3, exactly the three fixtures added |
| `goTranspiler nado` + `go build ./v4/` | exit 0 |
| `transpile.ts nado` + `ast.parse` sync/async | clean |

The randomised nonce does not destabilise the static suite: `skipKeys` in `ts/src/test/static/request/nado.json` already carries `nonce` and `signature`. I ran the JS suite three times specifically to confirm no flake was introduced.

Non-blocking notes inline.

Merge gate: 🟢
Merge probability: 95%

**Review by carlotestor — COMMENTED** [review]
Follow-up on `99b14d1` — this commit is docs + one fixture, no behaviour change.

The added `fetchOrders` case pins the archive cap end-to-end (`limit: 1000` in, `limit: 500` on the wire, `trigger.test.nado.xyz/v1/query`), which is exactly the gap I flagged: previously only the uncapped `limit: 10` path was covered. The widened `NotSupported` message now names `editOrderWs()` too, and both JSDocs document the refused `params.triggerPrice` / `stopPrice` / `stopLossPrice` / `takeProfitPrice` — consistent between `ts/src/nado.ts` and `ts/src/pro/nado.ts`.

The nonce comment is now accurate: `randNumber (6)` is a decimal-digit count, so the entropy is up to 999999 and stays inside the 2^20 = 1048576 low-bit window the shift reserves — no carry into the timestamp bits. Saying "a random value on the low bits" instead of "20 random low bits" is the right correction.

Nothing new to raise; the earlier verification still holds for this delta since no executable path changed outside the added fixture.

Merge gate: 🟢

File: rust/ccxt-base/src/runtime.rs [hunk 4]
```diff
@@ -717,14 +717,21 @@ pub use crate::precise::Precise;
...
     pub fn min(a: &Value, b: &Value) -> Value {
+        if both_ints(a, b) { return Value::Int(as_i64(a).unwrap().min(as_i64(b).unwrap())); }
         match (as_f64(a), as_f64(b)) { (Some(x), Some(y)) => Value::Float(x.min(y)), _ => Value::Null }
     }
     pub fn max(a: &Value, b: &Value) -> Value {
+        if both_ints(a, b) { return Value::Int(as_i64(a).unwrap().max(as_i64(b).unwrap())); }
         match (as_f64(a), as_f64(b)) { (Some(x), Some(y)) => Value::Float(x.max(y)), _ => Value::Null }
     }
     pub fn abs(a: &Value) -> Value {
+        if let Value::Int(n) = a { return Value::Int(n.abs()); }
```
Comment (carlotestor) [inline rust/ccxt-base/src/runtime.rs]: `n.abs()` on `i64::MIN` panics in debug builds and wraps in release, whereas the old `as_f64` path returned a finite `Float`. Unlikely to be reached from transpiled exchange code, but `n.wrapping_abs()` or a `checked_abs()` fallback to the float branch would keep this total.

**Review by carlotestor — COMMENTED** [review]
Follow-up on `bf9eb9a` — this commit is Rust-runtime only (`rust/ccxt-base/src/runtime.rs`); no change to `ts/src/nado.ts`, `ts/src/pro/nado.ts` or the static fixture since `99b14d1`.

The change makes `Math::min` / `Math::max` / `Math::abs` integer-preserving, which is the correct fix for the port-level half of the archive cap: `Math.min (limit, 500)` in `nado.ts` was landing in the Rust port as `Value::Float`, so the request body serialized `limit: 500.0` and a strict `u32` field rejects it. `add` / `sub` / `mul` in the same file already gate on `both_ints`, so this closes an inconsistency rather than inventing a new rule — and it reuses the existing private helper instead of duplicating the match.

Float semantics are preserved on mixed operands (`Int` + `Float` still goes through `as_f64`), and `Null` propagation is unchanged.

### Verification on `bf9eb9a`

| check | result |
|---|---|
| `cargo test -p ccxt-base math_tests` | 2 passed, 0 failed |
| mixed Int/Float and Null cases | covered by the added `min_max_abs_keep_float_semantics` |

One non-blocking edge case inline.

Merge gate: 🟢

**Review by carlotestor — COMMENTED** [review]
Follow-up on `99b14d1` — the branch was force-pushed back to this commit; `bf9eb9a` (the `rust/ccxt-base/src/runtime.rs` `Math::min`/`max`/`abs` integer-preservation change) is no longer in the head. Comparing `bf9eb9a...99b14d1` reports `behind by 1`, `files: []`, so nothing else moved: `ts/src/nado.ts`, `ts/src/pro/nado.ts` and `ts/src/test/static/request/nado.json` are byte-identical to the head I reviewed at 16:52 UTC.

So this head is exactly the state I already signed off on, minus the Rust-runtime fix. That is a reasonable split — the `Math::min` integer-preservation issue is a port-level concern that does not belong in a nado REST bugfix — but it does mean the `Math.min (limit, 500)` caps here still serialize as `limit: 500.0` in the Rust port until that change lands separately. Not a blocker for this PR (JS/Python/Go/C# are unaffected), just worth tracking so the archive-cap fixture does not fail Rust CI later.

Nothing new to raise on the TypeScript delta; the earlier verification stands unchanged.

Merge gate: 🟢

**Comment by carlosmiei** [comment]
@rayBastard did you manage to run a live test here?

**Comment by rayBastard** [comment]
> @rayBastard did you manage to run a live test here?

@carlosmiei 

Yes, ran it against testnet:

trigger order with triggerDirection 'up' accepted by the venue
two createOrder calls fired in the same millisecond via Promise.all - both accepted 
fetchCanceledOrders() / fetchCanceledAndClosedOrders() now work through the documented signature and returned the canceled trigger orders
fetchOrders(..., limit=1000, {trigger: true}) and fetchOHLCV(..., limit=1000) - server accepted the capped limit: 500 (500 candles returned)
All trigger orders canceled, none left open
